### PR TITLE
Improve NullUser with implementation of UserEntityInterface

### PR DESCRIPTION
**Pull request by AurelienPillevesse**
Add implementation of ```UserEntityInterface``` for ```NullUser``` to be aligned with OAuth2-Server library

### CODE REVIEW

**Comment by chalasr** [comment]
Could you please provide some link pointing to the related change on league/oauth2-server?

**Comment by AurelienPillevesse** [comment]
There is no update from league/oauth2-server but this class implements everything except this little update

File: src/Security/User/NullUser.php [hunk 1]
```diff
@@ -4,15 +4,25 @@
...
  * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
  */
-final class NullUser implements UserInterface
+final class NullUser implements UserInterface, UserEntityInterface
 {
+    /**
+     * @psalm-mutation-free
+     */
+    public function getIdentifier(): string
```
Comment (maxhelias) [inline src/Security/User/NullUser.php:21]: Shouldn't we respect the interface declaration with no return type and return a null value? https://github.com/thephpleague/oauth2-server/blob/master/src/Entities/UserEntityInterface.php#L19C7-L19C7

Comment (maxhelias) [inline src/Security/User/NullUser.php:21]: Shouldn't we respect the interface declaration with no return type and return a null value? https://github.com/thephpleague/oauth2-server/blob/master/src/Entities/UserEntityInterface.php#L19C7-L19C7

**Review by maxhelias — COMMENTED** [review]
The `NullUser` is related to the new security implementation https://github.com/thephpleague/oauth2-server-bundle/issues/34

**Comment by X-Coder264** [comment]
The `NullUser` is supposed to be used only in the context of the Symfony security layer (as that's what it was added for in the first place - in order to be able to use the client credentials grant in Symfony apps as that grant does not have a real user associated with it). I'm not sure what exactly do you want to achieve by making it implement an unrelated non Symfony interface?

**Comment by chalasr** [comment]
As @X-Coder264 said, this is not desired as we have entities separated from security objects.
Thanks for proposing.

**Pull request closed without merging**